internal/backup_time.go

```diff
@@ -2,10 +2,23 @@ package internal
 
 import "time"
 
+type BackupTimeDenotation int
+
+const (
+	ModificationTime BackupTimeDenotation = iota
+	CreationTime
+	NoData
+)
+
 // BackupTime is used to sort backups by
 // latest modified time.
 type BackupTime struct {
 	BackupName  string    `json:"backup_name"`
 	Time        time.Time `json:"time"`
 	WalFileName string    `json:"wal_file_name"`
 }
+
+type BackupTimeSlice struct {
+	Data []BackupTime                   `json:"data"`
+	TimeDenotation BackupTimeDenotation `json:"time_denotation"`
```

Comment:
Why do we need to save this information? Why don't just sort the backups in place of usage?

Reply:
`BackupTimeSlice` used in functions like [WriteBackupList](https://github.com/wal-g/wal-g/blob/b37391044879631b630dab418b6f29377d2cfdd3/internal/backup_list_handler.go#L134) and we cannot use just one `TimeDenotation` here because of possible old backups.

Is it flexible to left  `BackupTimeSlice` in [GetBackupsWithTargetOrdered](https://github.com/wal-g/wal-g/blob/b37391044879631b630dab418b6f29377d2cfdd3/internal/backup.go#L418) and still use `[]BackupTime` elsewhere?

Reply:
> `BackupTimeSlice` used in functions like [WriteBackupList](https://github.com/wal-g/wal-g/blob/b37391044879631b630dab418b6f29377d2cfdd3/internal/backup_list_handler.go#L134) and we cannot use just one `TimeDenotation` here because of possible old backups.
> 
> Is it flexible to left `BackupTimeSlice` in [GetBackupsWithTargetOrdered](https://github.com/wal-g/wal-g/blob/b37391044879631b630dab418b6f29377d2cfdd3/internal/backup.go#L418) and still use `[]BackupTime` elsewhere?

If there are old backups without metadata we can fall back to the old time denotation logic for all of the backups, to avoid mixing up the logic.

WriteBackupList shouldn't explicitly depend on the TimeDenotation in order to conform with OOP Open/Closed principle. Actually, we may simplify the logic and output the both `last_modified` and `created` fields in the output.